tests/ssr.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react'
+import { act, screen } from '@testing-library/react'
+import { renderToString } from 'react-dom/server'
+import { describe, expect, it, vi } from 'vitest'
+import { create } from 'zustand'
+
+interface BearStoreState {
+  bears: number
+}
+
+interface BearStoreAction {
+  increasePopulation: () => void
+}
+
+const initialState = { bears: 0 }
+const useBearStore = create<BearStoreState & BearStoreAction>((set) => ({
+  ...initialState,
+  increasePopulation: () => set(({ bears }) => ({ bears: bears + 1 })),
+}))
+
+function Counter() {
+  const { bears, increasePopulation } = useBearStore(
+    ({ bears, increasePopulation }) => ({
+      bears,
+      increasePopulation,
+    })
+  )
+
+  useEffect(() => {
+    increasePopulation()
+  }, [increasePopulation])
+
+  return <div>bears: {bears}</div>
+}
+
+describe.skipIf(!React.version.startsWith('18'))(
```

Comment:
Oh, I didn't know about `skipIf`!

We should use it in `jotai` too:
https://github.com/pmndrs/jotai/blob/d6afb777705eb3a90385817fd6db30db101f6caa/tests/react/transition.test.tsx#L12C1-L16

Free free to open a PR there.

Reply:
I am also glad to have learned about skipIf, and I will apply refactoring accordingly in Jotai as well

Reply:
I guess we are good to merge the changes :D. BTW, good job.

Reply:
Thank you so much for your review. 
I'm delighted to learn about the useful functions of vitest. I'll continue to check if there are any opportunities for further contributions.